File: kiosk-app/lib/models/table.dart
class KioskTable {
  final int id;
  final int tableNumber;
  final String? deviceId;
  final String status;

  KioskTable({
    required this.id,
    required this.tableNumber,
    this.deviceId,
    required this.status,
  });

  factory KioskTable.fromJson(Map<String, dynamic> json) => KioskTable(
        id: json['id'] as int,
        tableNumber: json['tableNumber'] as int,
        deviceId: json['deviceId'] as String?,
        status: json['status'] as String,
      );
}

File: kiosk-backend/src/main/kotlin/com/kiosk/domain/model/Table.kt
package com.kiosk.domain.model

enum class TableStatus {
    AVAILABLE,
    OCCUPIED,
    RESERVED
}

data class KioskTable(
    val id: Long = 0,
    val tableNumber: Int,
    val deviceId: String? = null,
    val status: TableStatus = TableStatus.AVAILABLE,
    val isActive: Boolean = true
)
